### PR TITLE
Rockchip: various fixes and enhancements

### PR DESCRIPTION
**Pull request by knaerzche**
It turned out our DRM patches are fine for HDMI 4k@60Hz usage but totally borked for any other output channel, such as DSI. This is now fixed up and even dual screen usage works fine (not really relevant for kodi) while keeping the HDMI achievements. 
I also squashed the drm patches, as some where overriding each other and dropped some unnecessary. Also enabled all panel drivers which might be relevant.

Though some recent kernel-side change (bug?) i2c might not correclty get initalized for RK3399, if it isn't done by u-boot for some boards. This is only relevant for the legacy boot-chain where we are using rk3399-evb configuration which initalizes i2c in SPL, which we don't use for the legacy boot chain. I added a fix for that.

Besides some kernel configuration tweaks and cleanups.

### CODE REVIEW

File: projects/Rockchip/devices/RK3288/linux/default/linux.arm.conf [hunk 1]
```diff
@@ -5752,7 +5752,7 @@ CONFIG_GENERIC_PHY=y
 CONFIG_PHY_ROCKCHIP_DP=m
 # CONFIG_PHY_ROCKCHIP_DPHY_RX0 is not set
 CONFIG_PHY_ROCKCHIP_EMMC=m
-CONFIG_PHY_ROCKCHIP_INNO_HDMI=m
+# CONFIG_PHY_ROCKCHIP_INNO_HDMI is not set
```
Comment (jernejsk) [inline projects/Rockchip/devices/RK3288/linux/default/linux.arm.conf:5755]: Is this intentional in this commit? You only talk about enabling USB.

Comment (knaerzche) [reply projects/Rockchip/devices/RK3288/linux/default/linux.arm.conf:5755]: Yes.
Inno HDMI phy is only used on RK322x and RK3328 and wouldn't want to make an extra commit for every change

**Review by jernejsk — APPROVED** [review]

**Comment by diederikdehaas** [comment]
Q: What's the reason ``CONFIG_DRM_PANEL_*`` are ``=y`` and not ``=m``?  
A quick scan through ``drivers/gpu/drm/panel/Kconfig`` only showed *tristate* (apart from ``CONFIG_DRM_PANEL`` itself).

(Context: wondering what to do for Debian kernel and DSI connector on Quartz64 boards)

**Comment by knaerzche** [comment]
> Q: What's the reason CONFIG_DRM_PANEL_* are =y and not =m?

For LE it's necessary to have them built-in as we are using an intramfs w/o support for loading modules during startup in order to display the splash screen. It might differ for debian.

**Comment by diederikdehaas** [comment]
I had a feeling that was the case. Thanks for explaining :+1:

**Pull request merged**